docs/blog/2019-11-10-how-to-convert-an-existing-gatsby-blog-to-use-mdx/index.md

```diff
@@ -0,0 +1,211 @@
++             title
++             description
++             siteUrl
++           }
++         }
++       }
++     `,
++     feeds: [
++       {
++         serialize: ({ query: { site, allMdx } }) =>
++           allMdx.edges.map(edge =>
++             Object.assign({}, edge.node.frontmatter, {
++               description: edge.node.excerpt,
++               data: edge.node.frontmatter.date,
++               url: site.siteMetadata.siteUrl + edge.node.fields.slug,
++               guid: site.siteMetadata.siteUrl + edge.node.fields.slug,
++               custom_elements: [{ "content:encoded": edge.node.body }],
++             })
++           ),
++         query: `
++         {
++           allMdx(
++             limit: 1000,
++             sort: { order: DESC, fields: [frontmatter___date] },
++           ) {
++             edges {
++               node {
++                 fields { slug }
++                 frontmatter {
++                   title
++                   date
++                 }
++                 body
++               }
++             }
++           }
++         }
++         `,
++         output: `/rss.xml`,
++         title: `Gatsby RSS feed`,
++       },
++     ],
++   },
++ },
+```
+
+### Step 3
+
+In `gatsby-node.js`, change `allMarkdownRemark` to `allMdx` and `MarkdownRemark` to `Mdx`.
+
+```diff:title=gatsby-node.js
+# line 11
+- allMarkdownRemark(
++ allMdx(
+
+# line 35
+- const posts = result.data.allMarkdownRemark.edges
++ const posts = result.data.allMdx.edges
+
+# line 56
+- if (node.internal.type === `MarkdownRemark`) {
++ if (node.internal.type === `Mdx`) {
+```
+
+### Step 4
+
+In `src/pages/index.js`, replace `allMarkdownRemark` with `allMdx` in both the `render()` method and also the GraphQL query.
+
+```diff:title=src/pages/index.js
+# line 13
+- const posts = data.allMarkdownRemark.edges
++ const posts = data.allMdx.edges
+
+# line 59
+- allMarkdownRemark(sort: { fields: [frontmatter___date], order: DESC }) {
++ allMdx(sort: { fields: [frontmatter___date], order: DESC }) {
+```
+
+### Step 5
+
+In `src/templates/blog-post.js`, add an import statement for `MDXRenderer` from `gatsby-plugin-mdx` and replace `markdownRemark` with `mdx` in both the `render()` method and also the GraphQL query. Also, we'll be using `body` instead of `html` in our GraphQL query.
+
+```diff:title=src/templates/blog-post.js
+# line 3
++ import { MDXRenderer } from "gatsby-plugin-mdx"
+
+# line 11
+- const post = this.props.data.markdownRemark
++ const post = this.props.data.mdx
+
+# line 41
+- <section dangerouslySetInnerHTML={{ __html: post.html }} />
++ <MDXRenderer>{post.body}</MDXRenderer>
+
+# line 93
+- markdownRemark(fields: { slug: { eq: $slug } }) {
++ mdx(fields: { slug: { eq: $slug } }) {
+
+# line 96
+- html
++ body
+```
+
+And... that's it! After these changes, a Gatsby blog should be able to use MDX files to render JSX with markdown! To test, add a `.mdx` file to `content/blog/` and check if the JSX renders.
```

Comment:
Should note which subfolder this would go in? `src`? `pages`? `components`?

Reply:
it would be in the root folder - full path is: `starters/blog/content/blog/mdx/index.mdx` or `<starterroot>`+`content/blog/mdx/index.mdx`

Reply:
Wonderful! But it's not for my information, it's for the readers. We want to allow anyone to use these instructions, and they are incredibly explicit throughout, which is great. Just want that reflected in this step as well.

Reply:
Ah ok - since all paths in doc should start from the root path without a beginning slash I assumed it is clear for the reader, but you are right: more explicit is always better

Reply:
And it doesn't have to be a full path, but a note that it's in the root directory would be great.

Reply:
done